packages/docusaurus-theme-classic/src/theme/Tabs/index.tsx

```diff
@@ -117,6 +142,18 @@ function TabsComponent(props: Props): JSX.Element {
       blockElementScrollPositionUntilNextRender(newTab);
       setSelectedValue(newTabValue);
 
+      const searchParams = new URLSearchParams(window.location.search);
+      const prevSearchParams = searchParams.get('tabs');
+      const prevVal = prevSearchParams ? JSON.parse(prevSearchParams) : null;
+      const newVal = {[groupId || NON_GROUP_TAB_KEY]: newTabValue};
+      const url = new URL(window.location.origin + window.location.pathname);
+      searchParams.set(
+        'tabs',
+        JSON.stringify(prevVal ? {...prevVal, ...newVal} : newVal),
+      );
+      url.search = searchParams.toString();
```

Comment:
I'd rather move any code to a custom hook as long as it's possible to create a good abstraction. Here it's the case for me.

https://kyleshevlin.com/use-encapsulation


> which may be overkill compared to just going with a couple lines of code

I don't see how a function call is overkill here. We could implement a whole app in a single file / function if we wanted, and it's not really overkill to split it into well-encapsulated concerns so that you are able to browse your app at different levels of abstractions

See also Clean code rule - Single level of abstraction